generators/app/templates/src/StarterKit/Startup/Startup.cs

```diff
@@ -1,114 +1,178 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
 using AutoMapper;
 using Digipolis.ApplicationServices;
 using Digipolis.Correlation;
-using Digipolis.Web;
-using Digipolis.Web.Startup;
-//--dataaccess-startupImports--
+using Digipolis.DataAccess;
 using Microsoft.AspNetCore.Builder;
-using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.AspNetCore.Rewrite;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+using Microsoft.EntityFrameworkCore.Migrations;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
-using StarterKit.Options;
+using Microsoft.OpenApi.Models;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Serialization;
+using StarterKit.DataAccess;
+using StarterKit.DataAccess.Options;
+using StarterKit.Framework.Logging;
+using StarterKit.Shared.Extensions;
+using StarterKit.Shared.Options;
 using StarterKit.Shared.Swagger;
-using Swashbuckle.AspNetCore.Swagger;
-using System;
-using System.IO;
-using System.Reflection;
 
-namespace StarterKit
+namespace StarterKit.Startup
 {
   public class Startup
   {
-    public Startup(IConfiguration configuration, IHostingEnvironment env)
+    public Startup(IConfiguration configuration, IHostEnvironment env)
     {
+      ApplicationBasePath = env.ContentRootPath;
       Configuration = configuration;
+      Environment = env;
     }
 
-    public IConfiguration Configuration { get; private set; }
-    public string ApplicationBasePath { get; private set; }
+    public IConfiguration Configuration { get; }
+    public string ApplicationBasePath { get; }
+    public IHostEnvironment Environment { get; }
+    private string XmlCommentsPath
+    {
+      get
+      {
+        var fileName = typeof(Startup).GetTypeInfo().Assembly.GetName().Name + ".xml";
+        return Path.Combine(ApplicationBasePath, fileName);
+      }
+    }
 
-    public void ConfigureServices(IServiceCollection services)
+    public virtual void ConfigureServices(IServiceCollection services)
```

Comment:
Waarom staat ConfigureServices virtual ? Moet dit ergens overridden worden ?

Reply:
Dit geeft de optie om makkelijk delen van startup te overschrijven in IntegrationTests.

Reply:
Klopt, wordt overridden bij het opstarten van de IntegrationTests